### DISCUSSION TITLE
[fieldArray] how I can fully replace the fields?

### DISCUSSION THREAD

**Discussion by OlehDutchenko** [react-hook-form/react-hook-form#6227] [Q&A] [answered]
I need to replace the fields programmatically.
My simplified code sample:

```ts
const { control } = useForm();
const { fields, remove, append } = useFieldArray({ control, name: 'demo' });

const handleFullyReplacement = (newFields: { value: string }[]): void => {
    remove(); // clear all prev data
    append(newFields); // set fully new
}

...
```

And this do not work. Docs says "You can not call actions one after another. Actions need to be triggered per render." (https://react-hook-form.com/api/usefieldarray/)

Has anyone ideas how to do that?

**Comment by bluebill1049** [comment, marked as answer]
`reset()` ?

**Reply by OlehDutchenko** [reply]
It was so easy... 
But not entirely obvious. Resetting the form is a slightly different task and semantic action. The current decision is more like a crutch, such a rough decision for the lack of another option. 

I understand this is probably due to technical processes. But it seems to me that a declarative wrapper can be created over this approach. I shall send PR if you do not mind

**Reply by OlehDutchenko** [reply]
I am found another way to do that from the underhood. PR is ready #6304
